force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -214,8 +251,16 @@ public with sharing class ClassesObjectsMethods {
     public static Integer calculateYearDifference(Date d1, Date d2) {
 
         // Subtract the difference in years
-
-        return null; // Replace null with the variable you used to store the result
+       
+        if(d1 == null || d2 == null){
+            return null;
+        }else{
+            Integer d1Year = d1.year();
+            Integer d2Year = d2.year();
+    
+            Integer yearDifference = Math.abs(d2Year - d1Year);
+            return yearDifference; 
+        } 
```

Comment:
no need to use else before you run this code. You already called `return null` above, so if the top part is executed the rest will not be.